### DISCUSSION TITLE
plotting IPF color key

### DISCUSSION THREAD

**Discussion by Patcurran** [MechMicroMan/DefDAP#111] [General]
can you plot the IPF color key?

**Comment by hakonanes** [comment]
Hi @Patcurran,

I believe the *m-3m* and *6/mmm* IPF color keys in orix (see [this tutorial](https://orix.readthedocs.io/en/stable/tutorials/inverse_pole_figures.html)) match the colors returned by DefDap's `Quat.calcIPFcolours()` for "cubic" and "hexagonal" symmetry, respectively. I believe this is the case because [the algorithm](https://github.com/MechMicroMan/DefDAP/blob/07b3aca47d7a67611c69d6e1996447f828e8137e/defdap/quat.py#L839) in DefDap is adapted from [the algorithm](https://github.com/BYU-MicrostructureOfMaterials/OpenXY/blob/3167b64b5a075a1f722160b7de9f39ed5873b6b5/Code/IPF_rgbcalc.m) in OpenXY. The algorithm in OpenXY says it mimicks the IPF colors in EDAX' OIM Analysis. The IPF color keys in orix do the same (they are adapted from MTEX' `ipfTSLkey`).

So, after [installing orix](https://orix.readthedocs.io/en/stable/user/installation.html), you can run:

```python
from orix import plot
ckey_m3m = plot.IPFColorKeyTSL("m-3m")
fig = ckey_m3m.plot(return_figure=True)  # Returns a Matplotlib figure
fig.savefig("ckey_m3m.png")
```

**Reply by mikesmic** [reply]
Thanks @hakonanes, yes the IPF colouring in defdap is based on openXY which is based on EDAX's colouring. So using orix is definitely the best way to get a nice colour key. @Patcurran you could play about with making an inset axis to display the key with your maps https://matplotlib.org/stable/gallery/axes_grid1/inset_locator_demo.html

**Reply by hakonanes** [reply]
Thank you for confirming, @mikesmic.

Note that we've wrapped Matplotlib's inset axis functionality in orix in an extra "projection". See the final part of [this example](https://orix.readthedocs.io/en/stable/examples/plotting/subplots.html).

**Reply by Patcurran** [reply]
thank you both
